### PR TITLE
Refactor job resource

### PR DESCRIPTION
**Pull request by Anatw**
- Add args to the docstring.
- Allow background propagation policy in delete().

Signed-off-by: Anat Wax <awax@redhat.com>

### CODE REVIEW

**Comment by Anatw** [comment]
!verified

File: ocp_resources/job.py [hunk 1]
```diff
@@ -63,3 +82,24 @@ def to_dict(self):
...
+        Args:
+            wait (bool): True to wait for Job to be deleted.
+            timeout (int): Time to wait for resource deletion.
+            body (dict): Content to send to delete().
+
+        Returns:
+            bool: True if delete succeeded, False otherwise.
+        """
+        if self.background_propagation_policy:
+            super().delete(
```
Comment (myakove) [inline ocp_resources/job.py:99]: Missing return.

File: ocp_resources/job.py [hunk 2]
```diff
@@ -63,3 +82,24 @@ def to_dict(self):
...
+                wait=wait,
+                timeout=timeout,
+                body=kubernetes.client.V1DeleteOptions(propagation_policy="Background"),
+            )
+        else:
+            super().delete()
```
Comment (myakove) [inline ocp_resources/job.py:105]: Missing return.

File: ocp_resources/job.py [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+
 from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.resource import NamespacedResource
 
 
 class Job(NamespacedResource):
     """
     Job object.
+
+        Args:
```
Comment (myakove) [inline ocp_resources/job.py:11]: Please fix indentation.

File: ocp_resources/job.py [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
+            client (DynamicClient): Dynamic client for connecting to a remote cluster.
+            teardown (bool): Indicates if this resource would need to be deleted.
+            privileged_client (DynamicClient): Instance of Dynamic client.
+            yaml_file (str): Yaml file for the resource.
+            delete_timeout (int): Timeout associated with delete action.
+            backoff_limit (int): The number of retries for a job.
+            restart_policy (str): The restart policy of the pod.
+            service_account (str): Optional. Service account name.
+            containers (list): List of containers belonging to the pod that the job will create.
+            background_propagation_policy (bool): If true - change propagation policy to 'Background'.
```
Comment (myakove) [inline ocp_resources/job.py:23]: `change propagation policy to 'Background'.` do we have more option other than `Background`? 
If so we need to get it here and not bool
Also please explain what is this and not just that we change it

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/job.py [hunk 5]
```diff
@@ -63,3 +82,24 @@ def to_dict(self):
                     self.res["spec"]["template"]["spec"][
                         "restartPolicy"
                     ] = self.restart_policy
+
+    def delete(self, wait=False, timeout=TIMEOUT_4MINUTES, body=None):
+        """
+        Delete Job
```
Comment (rnetser) [inline ocp_resources/job.py:88]: This is not a delete job; please change.

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/job.py (hunk 5 above)
Comment (Anatw) [reply ocp_resources/job.py:88]: Will change to "Delete Job object".

File: ocp_resources/job.py (hunk 4 above)
Comment (Anatw) [reply ocp_resources/job.py:23]: Ack.

File: ocp_resources/job.py (hunk 1 above)
Comment (Anatw) [reply ocp_resources/job.py:99]: Ack.

File: ocp_resources/job.py (hunk 2 above)
Comment (Anatw) [reply ocp_resources/job.py:105]: ack.

**Comment by Anatw** [comment]
!verified

**Comment by Anatw** [comment]
!verified

**Review by rnetser — DISMISSED** [review]

File: ocp_resources/job.py [hunk 6]
```diff
@@ -63,3 +84,26 @@ def to_dict(self):
...
+        if self.background_propagation_policy:
+            return super().delete(
+                wait=wait,
+                timeout=timeout,
+                body=kubernetes.client.V1DeleteOptions(
+                    propagation_policy=self.background_propagation_policy
+                ),
+            )
+        else:
+            return super().delete()
```
Comment (myakove) [inline ocp_resources/job.py:109]:
```
else:
            return super().delete()
```

You ignore `wait` and `timeout` here

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/job.py (hunk 6 above)
Comment (Anatw) [reply ocp_resources/job.py:109]: Ack

**Comment by Anatw** [comment]
!verified

File: setup.cfg [hunk 7]
```diff
@@ -1,13 +1,13 @@
 [metadata]
```
Comment (myakove) [inline setup.cfg:1]: This shouldn't be part of the PR.

**Review by myakove — CHANGES_REQUESTED** [review]

**Comment by Anatw** [comment]
!verified

**Review by myakove — APPROVED** [review]

File: ocp_resources/job.py [hunk 8]
```diff
@@ -63,3 +84,24 @@ def to_dict(self):
...
+
+        Args:
+            wait (bool): True to wait for Job to be deleted.
+            timeout (int): Time to wait for resource deletion.
+            body (dict): Content to send to delete().
+
+        Returns:
+            bool: True if delete succeeded, False otherwise.
+        """
+        body = (
```
Comment (rnetser) [inline ocp_resources/job.py:100]: this overwrites `body` which is passed to the function

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/job.py (hunk 8 above)
Comment (Anatw) [reply ocp_resources/job.py:100]: ack

**Comment by Anatw** [comment]
!verified

**Review by myakove — APPROVED** [review]

File: ocp_resources/job.py [hunk 9]
```diff
@@ -63,3 +84,25 @@ def to_dict(self):
...
+            bool: True if delete succeeded, False otherwise.
+        """
+        if not body:
+            body = (
+                kubernetes.client.V1DeleteOptions(
+                    propagation_policy=self.background_propagation_policy
+                )
+                if self.background_propagation_policy
+                else None
+            )
```
Comment (rnetser) [inline ocp_resources/job.py:107]:
```suggestion
        if not body and self.background_propagation_policy:
            ......
```

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/job.py (hunk 9 above)
Comment (Anatw) [reply ocp_resources/job.py:107]: ack

**Comment by Anatw** [comment]
!verified

**Review by myakove — APPROVED** [review]
